src/main/java/net/rptools/maptool/client/ui/MapPropertiesDialog.java

```diff
@@ -594,37 +594,42 @@ public void actionPerformed(ActionEvent e) {
       return button;
     }
 
-    // private JButton createClearButton() {
-    // JButton button = new JButton("Clear");
-    //
-    // return button;
-    // }
+    private JButton createClearButton() {
+      JButton button = new JButton("Clear");
+      button.addActionListener(
+          e -> {
+            selectedAsset = null;
+            setVisible(false);
+          });
+      return button;
+    }
 
     private JButton createOKButton() {
       JButton button = new JButton("OK");
-      button.addActionListener(
-          new ActionListener() {
-            public void actionPerformed(ActionEvent e) {
-              setVisible(false);
-            }
-          });
+      button.addActionListener(e -> setVisible(false));
       return button;
     }
 
     private JButton createCancelButton() {
       JButton button = new JButton("Cancel");
       button.addActionListener(
-          new ActionListener() {
-            public void actionPerformed(ActionEvent e) {
-              selectedAsset = null;
-              setVisible(false);
-            }
+          e -> {
+            selectedAsset = originalAsset;
+            setVisible(false);
           });
       return button;
     }
 
-    public Asset chooseAsset() {
+    /**
+     * Open the dialog and allow the user to select an asset for the map.
+     *
+     * @param asset - The currently selected asset.
+     * @return The selected asset. If "Clear" was clicked, returns null.
+     */
+    public Asset chooseAsset(Asset asset) {
+      originalAsset = asset;
       setVisible(true);
+      originalAsset = null;
```

Comment:
Why the double assignment?

Reply:
I did this to allow the previous asset to be garbage collected once it's been swapped or cleared. Might not actually make a difference depending on when the whole dialog gets GC'd though.

Reply:
I understand that, just not getting why putting to a value and then to null later. My understanding is that you are going to use originalAsset to restore the status of this in case the user hits the cancel button. However, at the end of this method originalAsset is always null.

So this will work if you selected a map and then hit the clearSelection, or if you had a map with an image selected and you want to remove it, but not in the case that you have a map with an image selected, select a new image, and then hit the cancel button: in that case I expect that the selectedAsset will become null.

I can't test it now, will do it later.

Reply:
Also see my comments on the issue.